src/libraries/System.Net.Quic/src/System/Net/Quic/QuicConnection.cs

```diff
@@ -422,8 +422,15 @@ public async ValueTask<QuicStream> OpenOutboundStreamAsync(QuicStreamType type,
 
             // Propagate ODE if disposed in the meantime.
             ObjectDisposedException.ThrowIf(_disposed == 1, this);
+
+            // In case of an incoming race when the connection is closed by the peer just before we open the stream,
+            // we receive QUIC_STATUS_ABORTED from MsQuic, but we don't know how the connection was closed. To
+            // distinguish this case, we throw ConnectionAborted without ApplicationErrorCode. In such a case, we
+            // can expect the connection close exception to be already reported on the connection level (or very soon).
+            bool connectionAbortedByPeer = ex is QuicException qe && qe.QuicError == QuicError.ConnectionAborted && qe.ApplicationErrorCode is null;
```

Comment:
Not strictly necessary, but putting this condition next to a place/code which creates this exception might be marginally better. As this is kind of a hack to propagate the info that we need to wait for connection SHUTDOWN_COMPLETE so I'd keep it somewhere together.